lib/Backend/arm64/LegalizeMD.cpp

```diff
@@ -315,7 +315,41 @@ IR::Instr * LegalizeMD::LegalizeLoad(IR::Instr *instr, uint opndNum, LegalForms
 
 void LegalizeMD::LegalizeIndirOffset(IR::Instr * instr, IR::IndirOpnd * indirOpnd, LegalForms forms, bool fPostRegAlloc)
 {
-   int32 offset = indirOpnd->GetOffset();
+    // For LEA, we have special handling of indiropnds
+    auto correctSize = [](IR::Instr* instr, IR::IndirOpnd* indirOpnd)
+    {
+        // If the base and index operands on an LEA are different sizes, grow the smaller one,
+        // matching the larger size but keeping the correct signedness for the type
+        IR::RegOpnd* baseOpnd = indirOpnd->GetBaseOpnd();
+        IR::RegOpnd* indexOpnd = indirOpnd->GetIndexOpnd();
+        if (   baseOpnd != nullptr
+            && indexOpnd != nullptr
+            && baseOpnd->GetSize() != indexOpnd->GetSize()
+            )
+        {
+            if (baseOpnd->GetSize() < indexOpnd->GetSize())
```

Comment:
I think we should assert(probably outside of the lambda) that baseOpnd is 64 bit. While it is not unreasonable to have a 32 bit index opnd for an LEA after doing 32 bit math, having a 32 base opnd on a 64 bit system is an error.

Reply:
Good point, added to updated version.